zkevm-circuits/src/root_circuit.rs

```diff
@@ -0,0 +1,197 @@
+                        .chain(accumulator_limbs)
+                        .collect_vec()
+                });
+            });
+
+        instances.map(|instances| Self {
+            svk: KzgSvk::<M>::new(params.get_g()[0]),
+            snark: SnarkWitness::new(
+                super_circuit_protocol,
+                super_circuit_instances,
+                super_circuit_proof,
+            ),
+            instances,
+        })
+    }
+
+    /// Returns accumulator indices in instance columns, which will be in
+    /// the last `4 * LIMBS` rows of instance column in `MainGate`.
+    pub fn accumulator_indices(&self) -> Vec<(usize, usize)> {
+        let offset = self.snark.protocol().num_instance.iter().sum::<usize>();
+        (offset..).map(|idx| (0, idx)).take(4 * LIMBS).collect()
+    }
+
+    /// Returns number of instance
+    pub fn num_instance(&self) -> Vec<usize> {
+        vec![self.snark.protocol().num_instance.iter().sum::<usize>() + 4 * LIMBS]
+    }
+
+    /// Returns instances
+    pub fn instances(&self) -> Vec<Vec<M::Scalar>> {
+        vec![self.instances.clone()]
+    }
+}
+
+impl<'a, M: MultiMillerLoop> Circuit<M::Scalar> for RootCircuit<'a, M> {
+    type Config = AggregationConfig;
+    type FloorPlanner = SimpleFloorPlanner;
+
+    fn without_witnesses(&self) -> Self {
+        Self {
+            svk: self.svk,
+            snark: self.snark.without_witnesses(),
+            instances: vec![M::Scalar::zero(); self.instances.len()],
+        }
+    }
+
+    fn configure(meta: &mut ConstraintSystem<M::Scalar>) -> Self::Config {
+        AggregationConfig::configure::<M::G1Affine>(meta)
+    }
+
+    fn synthesize(
+        &self,
+        config: Self::Config,
+        mut layouter: impl Layouter<M::Scalar>,
+    ) -> Result<(), Error> {
+        config.load_table(&mut layouter)?;
+        let (instances, accumulator_limbs) =
+            config.aggregate::<M>(&mut layouter, &self.svk, [self.snark])?;
+
+        // Constrain equality to instance values
+        let main_gate = config.main_gate();
+        for (row, limb) in instances
+            .into_iter()
+            .flatten()
+            .flatten()
+            .chain(accumulator_limbs)
+            .enumerate()
+        {
+            main_gate.expose_public(layouter.namespace(|| ""), limb, row)?;
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::{
+        root_circuit::{compile, Config, PoseidonTranscript, RootCircuit},
+        super_circuit::{super_circuit_tests::sampl_block, SuperCircuit},
+    };
+    use halo2_proofs::{
+        circuit::Value,
+        dev::MockProver,
+        halo2curves::bn256::Bn256,
+        plonk::{create_proof, keygen_pk, keygen_vk},
+        poly::kzg::{
+            commitment::{KZGCommitmentScheme, ParamsKZG},
+            multiopen::ProverGWC,
+        },
+    };
+    use itertools::Itertools;
+    use rand::rngs::OsRng;
+
+    #[ignore = "Due to high memory requirement"]
```

Comment:
Since it's trying to create a `SuperCircuit` proof with `k >= 18`, so I believe it requires a tons of memory to run, so I didn't bother to try enable this to test the CI machine.